### PR TITLE
Implements a StorageReader interface.

### PR DESCRIPTION
**Pull request by elgris**

### CODE REVIEW

File: storageconnector/storageconnector.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+type SearchParams struct {
+	Name   regexp.Regexp
+	Limit  int64
+	Offset int64
+}
+
+// FeatureMeta provides short information about the Feature.
+// This piece of data better to be indexed by a storage
+type FeatureMeta struct {
+	ID            string
```
Comment (nayef) [inline storageconnector/storageconnector.go]: we are not keeping any ID for feature. name is the unique key here

File: storageconnector/storageconnector.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	Name          string
+	Author        string
+	CreatedAt     time.Time
+	UpdatedAt     time.Time
+	DependencyIDs []string
```
Comment (nayef) [inline storageconnector/storageconnector.go]: dependencyNames maybe?

File: storageconnector/storageconnector.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	// SearchMeta returns an arbitrary ordered list of FeatureMeta records using given SearchParams
+	SearchMeta(SearchParams) ([]FeatureMeta, error)
+
+	// GetMeta returns a single FeatureMeta by given Feature ID. Meta is a small piece of data,
+	// so it should be indexed by a storage and accessed rather quickly.
+	GetMeta(featureID string) (FeatureMeta, error)
+
+	// Get returns a full feature data from a storage. This operation is a way slower than GetMeta, so for
+	// quick lookups GetMeta is better to be used.
+	Get(featureID string) (Feature, error)
```
Comment (nayef) [inline storageconnector/storageconnector.go]: featureName instead of featureID

File: storageconnector/storageconnector.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+	// Resolve resolves dependencies for a given Feature and returns an **ordered** list of Features.
+	// Ordering is critical here, it defines a way how Features should be executed.
+	//
+	// Example:
+	// If you're given a list [FeatureA, FeatureB, FeatureC, FeatureD] that might mean that FeatureC depends on
+	// FeatureB which in its turn depends on FeatureA. In given example FeatureD has no dependencies, but it must
+	// be executed **after** all dependencies for FeatureC are resolved.
+	Resolve(featureID string) ([]Feature, error)
```
Comment (nayef) [inline storageconnector/storageconnector.go]: featureName instead of featureID

**Comment by stoewer** [comment]
What about:
* `featureID` -> `name`
* `DependencyIDs` -> `Dependencies`

`featureName` and `DependencyNames` are unnecessarily long

**Comment by stoewer** [comment]
LGTM :+1:

**Pull request merged**